clients/go/admin/token_source_provider.go

```diff
@@ -134,14 +135,18 @@ type ClientCredentialsTokenSourceProvider struct {
 
 func NewClientCredentialsTokenSourceProvider(ctx context.Context, cfg *Config,
 	clientMetadata *service.PublicClientAuthConfigResponse, tokenURL string) (TokenSourceProvider, error) {
-
-	secretBytes, err := ioutil.ReadFile(cfg.ClientSecretLocation)
-	if err != nil {
-		logger.Errorf(ctx, "Error reading secret from location %s", cfg.ClientSecretLocation)
-		return nil, err
+	var secret string
+	if len(cfg.ClientSecretLocation) > 0 {
+		secretBytes, err := ioutil.ReadFile(cfg.ClientSecretLocation)
+		if err != nil {
+			logger.Errorf(ctx, "Error reading secret from location %s", cfg.ClientSecretLocation)
+			return nil, err
+		}
+		secret = strings.TrimSpace(string(secretBytes))
```

Comment:
Can we have this done in both the conditions

Reply:
done